### ISSUE TITLE
Adding connections using mongodb:// connection strings is broken

### ISSUE THREAD

**Issue by azec-pdx** [mongodb-js/vscode#169]
#### Current Behavior

Let's assume I start by creating new MongoDB Connection to my Replica with these parameters...

**Hostname:** `2.microservicedomain.mongo.us-east-1.my-aws-acct.local`
**Port:** `27017`
**Authentication:** `SCRAM-SHA-256`
**Username:** `myUsername`
**Password:** `C;Ib86n5b8{AnExew[TU%XZy,)E6G!dk`
**Authentication Database:** `admin`
**Replica Set Name:** `microservicedomain`

When I create connection by filling out form in VS Code, it connects fine.
If I now copy connection string from this connection (`Right click on connection` --> `Copy Connection String`) , it produces following connection string: `mongodb://myUsername:C%3BIb86n5b8%7BAnExew%5BTU%25XZy%2C)E6G!dk@2.microservicedomain.mongo.us-east-1.my-aws-acct.local:27017/?authSource=admin&authMechanism=SCRAM-SHA-256&replicaSet=microservicedomain&readPreference=primary&ssl=false`

This connection string is problematic in multiple ways:

* Password is not compliant with [rfc3986](https://tools.ietf.org/html/rfc3986#section-2.1) spec for percent-encoding that is used to sanitize MongoDB connection string usernames & passwords
* If I delete this connection added through UI and use this generated connection string to re-create same MongoDB connection, it fails. Therefore, connection string is not idempotent if I can not re-use it to create connection.
* If I try to feed this string to [mongosh](https://www.mongodb.com/blog/post/introducing-the-new-shell) it will not connect, again because Connection String is incorrect.

Additionally to this, creating the new Connection with valid Connection String which includes percent-encoded password is not working:

* Assume that I take valid Connection String where password was already correctly percent-encoded: `mongodb://myUsername:C%3BIb86n5b8%7BAnExew%5BTU%25XZy%2C%29E6G%21dk@2.microservicedomain.mongo.us-east-1.my-aws-acct.local:27017/?authSource=admin&authMechanism=SCRAM-SHA-256&replicaSet=microservicedomain&readPreference=primary&ssl=false` and feed it to VS Code in attempt to create New Connection from Connection String. **It fails to connect.** While this connection string does work with `mongosh` and classic `mongo shell` , it does not in VS Code plugin. 
* Assume that I take Connection String where password was NOT percent-encoded:
`mongodb://myUsername:C;Ib86n5b8{AnExew[TU%XZy,)E6G!dk@2.microservicedomain.mongo.us-east-1.my-aws-acct.local:27017/?authSource=admin&authMechanism=SCRAM-SHA-256&replicaSet=microservicedomain&readPreference=primary&ssl=false` and feed it to VS Code in attempt to create New Connection from Connection String. **It fails to connect**. While this connection string also doesn't work with `mongosh`, the point being made here is that creating connection in VS Code plugin from Connection Strings absolutely does not work, no matter what you pass!

#### Code/Gist

According to this ticket [PYTHON-1282](https://jira.mongodb.org/browse/PYTHON-1282) and resulting [Pull Request](https://github.com/mongodb/mongo-python-driver/commit/c72ee15b525501823965ca83fb8cc31dcbe2a542), even in MongoDB driver for Python, [urllib.parse.quote_plus](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote_plus) and [urllib.parse.unquote_plus](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.unquote_plus) are being used to percent-encode/decode usernames and passwords when creating MongoDB Connection Strings. I used this knowledge to quickly generate percent-encoded password: `C;Ib86n5b8{AnExew[TU%XZy,)E6G!dk` (raw) --> `C%3BIb86n5b8%7BAnExew%5BTU%25XZy%2C%29E6G%21dk` (percent-enc)

```python
# python                                            ✔  02:20:15 PM  ▓▒░
Python 3.7.4 (default, Dec 24 2019, 13:36:33)
[Clang 10.0.1 (clang-1001.0.46.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from urllib.parse import unquote_plus
>>> from urllib.parse import quote_plus
>>> mypass="C;Ib86n5b8{AnExew[TU%XZy,)E6G!dk"
>>> mypass_percent_enc=quote_plus(mypass)
>>> print(mypass_percent_enc)
C%3BIb86n5b8%7BAnExew%5BTU%25XZy%2C%29E6G%21dk
```

#### Expected Behavior/Code

I expect that once I create new MongoDB connection through VS Code plugin UI and then `Copy Connection String` (menu option) , it provides me with correct and idempotent Connection String which I can feed to any mongo shell (`mongosh` or classic), as well as if I want to re-create same connection but using Connection String. I don't want to fill out the UI connection elements again if for example my MongoDB credentials got rotated. Currently the way this is broken doesn't make it usable at all to re-create connections (e.g. in the event when password only has changed). 

#### Environment

- **node.js / npm versions:** [node `v10.5.0` / npm `6.14.4` ]
- **OS:** [OSX / macOS Catalilna / `Version 10.15.6 (19G73)` ]
- **python:** [Python `3.7.4`]
- **VS Code:**
   ```
   Version: 1.48.2
   Commit: a0479759d6e9ea56afa657e454193f72aef85bd0
   Date: 2020-08-25T10:09:08.021Z (3 wks ago)
   Electron: 7.3.2
   Chrome: 78.0.3904.130
   Node.js: 12.8.1
   V8: 7.8.279.23-electron.0
   ```
- **MongoDB for VS Code (plugin):** [`v0.11`]

#### Possible Solution

Make sure what VS Code plugin uses to sanitize connection strings is compliant with all other places where Connection Strings can be used.

**Comment by mmarcon** [comment]
This is fixed in v0.3.0, released on 2020/11/24.

**Comment by azec-pdx** [comment]
Thanks for the update @mmarcon !
